Mehdi zare/fmp data doc

MehdiZare opened the pull request:
Title: community: add Financial Modeling Prep (FMP) API integration

**Description:** Adding LangChain integration for Financial Modeling Prep (FMP) API to enable semantic search and structured tool creation for financial data endpoints. This integration provides semantic endpoint search using vector stores and automatic tool creation with proper typing and error handling. Users can discover relevant financial endpoints using natural language queries and get properly typed LangChain tools for discovered endpoints.

**Issue:** N/A

**Dependencies:**
- fmp-data>=0.3.1
- langchain-core>=0.1.0
- faiss-cpu
- tiktoken

**Twitter handle:** @mehdizarem

Unit tests and example notebook have been added:
1. Tests are in `tests/integration_tests/est_tools.py` and `tests/unit_tests/test_tools.py`
2. Example notebook is in `docs/tools.ipynb`

All format, lint and test checks pass:
```bash
pytest
mypy .
```

Dependencies are imported within functions and not added to pyproject.toml. The changes are backwards compatible and only affect the community package.

MehdiZare committed:
Add FMP Data integration tutorial notebook

This commit introduces a new tutorial notebook for using the `langchain-fmp-data` package. The notebook provides examples for accessing financial market data, performing queries, and integrating with LangChain agents for various financial analyses. It also covers advanced usage and customization options.

MehdiZare committed:
Add FMP Data integration tutorial notebook

This commit introduces a new tutorial notebook for using the `langchain-fmp-data` package. The notebook provides examples for accessing financial market data, performing queries, and integrating with LangChain agents for various financial analyses. It also covers advanced usage and customization options.

MehdiZare committed:
Merge remote-tracking branch 'origin/master'

MehdiZare committed:
"Refactor FMP data tutorial notebook format"

Converted the FMP data tutorial notebook to use percent-based formatting for cleaner and more compatible code representation. This update improves readability, simplifies version control, and aligns with standard Jupyter practices.

MehdiZare committed:
"Refactor FMP data tutorial notebook format"

Converted the FMP data tutorial notebook to use percent-based formatting for cleaner and more compatible code representation. This update improves readability, simplifies version control, and aligns with standard Jupyter practices.

MehdiZare committed:
Simplify and restructure FMPData integration tutorial

Replaced verbose and redundant notebook content with a cleaner, modular structure. Focused on providing clear setup instructions, concise component descriptions, and flexible usage examples for FMPDataToolkit and FMPDataTool.

MehdiZare committed:
Simplify and restructure FMPData integration tutorial

Replaced verbose and redundant notebook content with a cleaner, modular structure. Focused on providing clear setup instructions, concise component descriptions, and flexible usage examples for FMPDataToolkit and FMPDataTool.

MehdiZare committed:
Simplify FMPData integration notebook content.

Refactored the FMPData integration notebook by removing redundant metadata and restructuring it to align with a more concise cell format. This enhances readability while maintaining functionality and documentation clarity.

MehdiZare committed:
Simplify FMPData integration notebook content.

Refactored the FMPData integration notebook by removing redundant metadata and restructuring it to align with a more concise cell format. This enhances readability while maintaining functionality and documentation clarity.

MehdiZare committed:
Refactor FMPData notebook structure for improved readability

Reorganized the FMPData integration documentation by converting the notebook to use cell-based markdown and code blocks. This enhances clarity and maintainability, ensuring better formatting and structure for users.

MehdiZare committed:
Refactor FMPData notebook structure for improved readability

Reorganized the FMPData integration documentation by converting the notebook to use cell-based markdown and code blocks. This enhances clarity and maintainability, ensuring better formatting and structure for users.

MehdiZare committed:
Refactor FMPData notebook structure for clarity

Streamlined the FMPData integration notebook by modifying its structure, reorganizing sections, and adopting consistent cell syntax using delimiters. This enhances readability and maintainability of the documentation.

MehdiZare committed:
Refactor FMPData notebook structure for clarity

Streamlined the FMPData integration notebook by modifying its structure, reorganizing sections, and adopting consistent cell syntax using delimiters. This enhances readability and maintainability of the documentation.

MehdiZare committed:
Convert FMP data documentation notebook to markdown format

Switches the `.ipynb` file to a markdown-based format using delimited code blocks. Simplifies maintainability, improves readability, and aligns with standard documentation practices.

MehdiZare committed:
Convert FMP data documentation notebook to markdown format

Switches the `.ipynb` file to a markdown-based format using delimited code blocks. Simplifies maintainability, improves readability, and aligns with standard documentation practices.

MehdiZare committed:
Convert FMP data documentation notebook to markdown format

Switches the `.ipynb` file to a markdown-based format using delimited code blocks. Simplifies maintainability, improves readability, and aligns with standard documentation practices.

MehdiZare committed:
Convert FMP data documentation notebook to markdown format

Switches the `.ipynb` file to a markdown-based format using delimited code blocks. Simplifies maintainability, improves readability, and aligns with standard documentation practices.

MehdiZare committed:
Convert FMP data documentation notebook to markdown format

Switches the `.ipynb` file to a markdown-based format using delimited code blocks. Simplifies maintainability, improves readability, and aligns with standard documentation practices.

MehdiZare committed:
Convert FMP data documentation notebook to markdown format

Switches the `.ipynb` file to a markdown-based format using delimited code blocks. Simplifies maintainability, improves readability, and aligns with standard documentation practices.

MehdiZare committed:
Refactor FMPData integration documentation notebook

Reorganized the FMPData documentation content for clarity and improved formatting. Simplified the structure by converting cells to Markdown and code blocks, ensuring better readability and usability within LangChain integrations.

MehdiZare committed:
Refactor FMPData integration documentation notebook

Reorganized the FMPData documentation content for clarity and improved formatting. Simplified the structure by converting cells to Markdown and code blocks, ensuring better readability and usability within LangChain integrations.

MehdiZare committed:
Update FMPData integration notebook for consistency and clarity

Refactored the structure of the FMPData integration notebook. Simplified API examples and adjusted formatting for better readability. Streamlined variable names and aligned tool usage with best practices.

MehdiZare committed:
Update FMPData integration notebook for consistency and clarity

Refactored the structure of the FMPData integration notebook. Simplified API examples and adjusted formatting for better readability. Streamlined variable names and aligned tool usage with best practices.

MehdiZare committed:
Update FMPData integration notebook for consistency and clarity

Refactored the structure of the FMPData integration notebook. Simplified API examples and adjusted formatting for better readability. Streamlined variable names and aligned tool usage with best practices.

MehdiZare committed:
Update FMPData integration notebook for consistency and clarity

Refactored the structure of the FMPData integration notebook. Simplified API examples and adjusted formatting for better readability. Streamlined variable names and aligned tool usage with best practices.

MehdiZare committed:
Update FMPData integration notebook for consistency and clarity

Refactored the structure of the FMPData integration notebook. Simplified API examples and adjusted formatting for better readability. Streamlined variable names and aligned tool usage with best practices.

MehdiZare committed:
Update FMPData integration notebook for consistency and clarity

Refactored the structure of the FMPData integration notebook. Simplified API examples and adjusted formatting for better readability. Streamlined variable names and aligned tool usage with best practices.

MehdiZare committed:
Update FMPData integration notebook for consistency and clarity

Refactored the structure of the FMPData integration notebook. Simplified API examples and adjusted formatting for better readability. Streamlined variable names and aligned tool usage with best practices.

MehdiZare committed:
Update FMPData integration notebook for consistency and clarity

Refactored the structure of the FMPData integration notebook. Simplified API examples and adjusted formatting for better readability. Streamlined variable names and aligned tool usage with best practices.

MehdiZare committed:
Convert FMP integration documentation to Jupyter Notebook format

Reformatted the FMP integration guide into a streamlined Jupyter Notebook. This improves readability, supports direct code execution, and better integrates setup, usage, and examples for the LangChain FMP tools.

MehdiZare committed:
Convert FMP integration documentation to Jupyter Notebook format

Reformatted the FMP integration guide into a streamlined Jupyter Notebook. This improves readability, supports direct code execution, and better integrates setup, usage, and examples for the LangChain FMP tools.

MehdiZare committed:
"Refactor FMPData integration content in notebook format"

Converted the FMPData integration guide to a more structured notebook format with distinct cell types for better readability and usability. This includes updating raw and markdown cells for explanations and using code cells for execution examples and API references.

MehdiZare committed:
"Refactor FMPData integration content in notebook format"

Converted the FMPData integration guide to a more structured notebook format with distinct cell types for better readability and usability. This includes updating raw and markdown cells for explanations and using code cells for execution examples and API references.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

MehdiZare committed:
Convert FMPData documentation to Jupyter Notebook format

Revised the FMPData integration documentation by transitioning it from a standalone text format to a structured Jupyter Notebook format. This improves readability and interactivity, allowing users to execute code snippets directly within the document.

Marsman1996 committed:
community: add init for unstructured file loader (#29101)

## Description
Add `__init__` for unstructured loader of
epub/image/markdown/pdf/ppt/word to restrict the input type to `str` or
`Path`.
In the
[signature](https://python.langchain.com/api_reference/community/document_loaders/langchain_community.document_loaders.markdown.UnstructuredMarkdownLoader.html)
these unstructured loaders receive `file_path: str | List[str] | Path |
List[Path]`, but actually they only receive `str` or `Path`.

## Issue
None

## Dependencies
No changes.

ThomasSaulou committed:
fix chatperplexity: remove 'stream' from params in _stream method (#29173)

quick fix chatperplexity: remove 'stream' from params in _stream method

Priyansh121096 committed:
community: add missing format specifier in error log in CubeSemanticLoader (#29172)

Thank you for contributing to LangChain!

- [x] **PR title**: "package: description"
- Where "package" is whichever of langchain, community, core, etc. is
being modified. Use "docs: ..." for purely docs changes, "infra: ..."
for CI changes.
  - Example: "community: add foobar LLM"


- [x] **PR message**
- **Description:** Add a missing format specifier in an an error log in
`langchain_community.document_loaders.CubeSemanticLoader`
- **Issue:** raises `TypeError: not all arguments converted during
string formatting`


- [ ] **Add tests and docs**: If you're adding a new integration, please
include
1. a test for the integration, preferably unit tests that do not rely on
network access,
2. an example notebook showing its use. It lives in
`docs/docs/integrations` directory.


- [ ] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified. See contribution
guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional
ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in
langchain.

If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

syedmuneeb321 committed:
Fixed the import error in OpenAIWhisperParserLocal and resolved the L… (#29168)

…angChain parser issue.

Thank you for contributing to LangChain!

- [ ] **PR title**: "package: description"
- Where "package" is whichever of langchain, community, core, etc. is
being modified. Use "docs: ..." for purely docs changes, "infra: ..."
for CI changes.
  - Example: "community: add foobar LLM"


- [ ] **PR message**: ***Delete this entire checklist*** and replace
with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change
- **Twitter handle:** if your PR gets announced, and you'd like a
mention, we'll gladly shout you out!


- [ ] **Add tests and docs**: If you're adding a new integration, please
include
1. a test for the integration, preferably unit tests that do not rely on
network access,
2. an example notebook showing its use. It lives in
`docs/docs/integrations` directory.


- [ ] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified. See contribution
guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional
ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in
langchain.

If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

ccurme committed:
langchain[patch]: update extended tests for compatibility with langchain-openai==0.3 (#29174)

winsento committed:
Add additional_instructions on openai assistan runs create. (#29164)

- **Description**: In the functions `_create_run` and `_acreate_run`,
the parameters passed to the creation of
`openai.resources.beta.threads.runs` were limited.

  Source: 
  ```
  def _create_run(self, input: dict) -> Any:
        params = {
            k: v
            for k, v in input.items()
            if k in ("instructions", "model", "tools", "run_metadata")
        }
        return self.client.beta.threads.runs.create(
            input["thread_id"],
            assistant_id=self.assistant_id,
            **params,
        )
  ```
- OpenAI Documentation
([createRun](https://platform.openai.com/docs/api-reference/runs/createRun))

- Full list of parameters `openai.resources.beta.threads.runs` ([source
code](https://github.com/openai/openai-python/blob/main/src/openai/resources/beta/threads/runs/runs.py#L91))

 
- **Issue:** Fix #17574 



- [x] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified. See contribution
guidelines for more: https://python.langchain.com/docs/contributing/

Co-authored-by: ccurme <chester.curme@gmail.com>

ccurme commented on docs/docs/integrations/tools/fmp.ipynb:
```diff
@@ -0,0 +1,401 @@
+   "id": "62828c19159a0da8",
+   "metadata": {},
+   "source": [
+    "## Overview\n",
```
I'd add a title here, e.g.,
```
# Financial Modeling Prep

Access financial market data through FMP.

## Overview
...
```
Otherwise the title just renders as `fmp` (check out the docs preview for this page [here](https://langchain-git-fork-mehdizare-mehdizare-fmp-data-doc-langchain.vercel.app/docs/integrations/tools/fmp/)).

ccurme commented on docs/docs/integrations/tools/fmp.ipynb:
```diff
@@ -0,0 +1,401 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain.agents import AgentExecutor, create_openai_functions_agent\n",
```
I recommend demonstrating with `langgraph` as is done in the toolkit docs template [here](https://github.com/langchain-ai/langchain/blob/master/libs/cli/langchain_cli/integration_template/docs/toolkits.ipynb).

MehdiZare replied:
Thank you, @ccurme ! I just applied the requested changes. Please let me know if anything else needs to be updated.

ccurme reviewed (commented):
Thank you for this, looks great!

Before merging, we need to add `langchain-fmp-data` to the [package registry](https://github.com/langchain-ai/langchain/blob/master/libs/packages.yml). Is there a Github repo associated with this package? Its [pypi](https://pypi.org/project/langchain-fmp-data/) page just points to the LangChain Github.

Other tasks to do before merging:
- Address comments (two small ones);
- Undo changes to `poetry.lock`.

I'd also consider adding a [provider page](https://github.com/langchain-ai/langchain/tree/master/docs/docs/integrations/providers) for FMP, but this can be done in a separate PR if desired.

tymzar committed:
community: Fix rank-llm import paths for new 0.20.3 version (#29154)

# **PR title**: "community: Fix rank-llm import paths for new 0.20.3
version"
- The "community" package is being modified to handle updated import
paths for the new `rank-llm` version.

---

## Description
This PR updates the import paths for the `rank-llm` package to account
for changes introduced in version `0.20.3`. The changes ensure
compatibility with both pre- and post-revamp versions of `rank-llm`,
specifically version `0.12.8`. Conditional imports are introduced based
on the detected version of `rank-llm` to handle different path
structures for `VicunaReranker`, `ZephyrReranker`, and `SafeOpenai`.

## Issue
RankLLMRerank usage throws an error when used GPT (not only) when
rank-llm version is > 0.12.8 - #29156

## Dependencies
This change relies on the `packaging` and `pkg_resources` libraries to
handle version checks.

## Twitter handle
@tymzar

gccornejo441 committed:
docs: Fix old link to Unstructured package in document_loader_markdown.ipynb (#29175)

Fixed a broken link in `document_loader_markdown.ipynb` to point to the
updated documentation page for the Unstructured package.
Issue: N/A
Dependencies: None

Commit:
cookbook: fix typo in cookbook/mongodb-langchain-cache-memory.ipynb (#29149)

Description: fix "enviornment" into "environment". 
Issue: Typo
Dependencies: None
Twitter handle: zrwang01

NikhilShahi committed:
docs: add HyperbrowserLoader docs (#29143)

### Description
This PR adds docs for the
[langchain-hyperbrowser](https://pypi.org/project/langchain-hyperbrowser/)
package. It includes a document loader that uses Hyperbrowser to scrape
or crawl any urls and return formatted markdown or html content as well
as relevant metadata.
[Hyperbrowser](https://hyperbrowser.ai) is a platform for running and
scaling headless browsers. It lets you launch and manage browser
sessions at scale and provides easy to use solutions for any webscraping
needs, such as scraping a single page or crawling an entire site.

### Issue
None

### Dependencies
None

### Twitter Handle
`@hyperbrowser`

varjas committed:
langchain: support api key argument with OpenAI moderation chain (#29140)

**Description:** Makes it possible to instantiate
`OpenAIModerationChain` with an `openai_api_key` argument only and no
`OPENAI_API_KEY` environment variable defined.

**Issue:** https://github.com/langchain-ai/langchain/issues/25176

**Dependencies:** `openai`

---------

Co-authored-by: ccurme <chester.curme@gmail.com>

ccurme committed:
cli[patch]: fix anchor links in templates (#29178)

These are outdated and can break docs builds.

Commit:
docs: fix httpx conflicts with overrides in docs build (#29180)

Commit:
anthropic: pydantic mypy plugin (#29144)

MehdiZare committed:
Add new package entry for langchain-fmp-data

This commit adds a new package entry to `libs/packages.yml` for the `langchain-fmp-data` repository. This ensures it is now tracked and available in the list of packages.

MehdiZare committed:
Add new package entry for langchain-fmp-data

This commit adds a new package entry to `libs/packages.yml` for the `langchain-fmp-data` repository. This ensures it is now tracked and available in the list of packages.

MehdiZare committed:
Add documentation for FMP Data integration

Introduce initial docs for integrating with the FMP Data Python package. Includes installation steps, setup instructions, and a usage example.

MehdiZare committed:
Add documentation for FMP Data integration

Introduce initial docs for integrating with the FMP Data Python package. Includes installation steps, setup instructions, and a usage example.

MehdiZare committed:
revert to original

MehdiZare committed:
revert to original

MehdiZare committed:
Rename FMP integration files and fix minor docs typo

Renamed `fmp.ipynb` to `fmp-data.ipynb` for better clarity and updated the corresponding documentation. Also fixed a minor typo in the markdown header from "Toolsg" to "Tools".

MehdiZare committed:
Rename FMP integration files and fix minor docs typo

Renamed `fmp.ipynb` to `fmp-data.ipynb` for better clarity and updated the corresponding documentation. Also fixed a minor typo in the markdown header from "Toolsg" to "Tools".

MehdiZare committed:
Merge remote-tracking branch 'origin/MehdiZare/fmp-data-doc' into MehdiZare/fmp-data-doc

ccurme commented:
Thanks @MehdiZare, I think this picked up some merge or rebase issues (PR is currently changing 40 files). Can you make sure you are merging against latest master branch?

MehdiZare commented:
Sorry about it. I'll open a new PR. it get's corrupted.
